src/System.Private.CoreLib/src/System/Decimal.cs

```diff
@@ -56,9 +52,8 @@ namespace System
     // throw an exception. A conversion from float or double to
     // Decimal throws an OverflowException if the value is not within
     // the range of the Decimal type.
-    [StructLayout(LayoutKind.Sequential)]
     [Serializable]
-    [System.Runtime.Versioning.NonVersionable] // This only applies to field layout
+    [StructLayout(LayoutKind.Explicit)]
```

Comment:
This breaks unix PInvoke interop as @jkotas already mentioned.

Reply:
Sounds like there should be a test for it

Reply:
#18908